samples/basic/component_build.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import kfp
+from kfp import compiler
+from kfp import dsl
+
+
+def process_args():
+  """Define arguments and assign default values to the ones that are not set.
+
+  Returns:
+    args: The parsed namespace with defaults assigned to the flags.
+  """
+
+  parser = argparse.ArgumentParser(
+      description='Basic sample to demonstrate component build.')
+  parser.add_argument(
+      '--output_path',
+      default=None,
+      required=True,
+      help='GCS bucket path to store temporary assets from build process')
+  parser.add_argument(
+      '--target_image',
+      default='component_build:latest',
+      help='Name:tag for the image that will be built by component build')
+  parser.add_argument(
+      '--container_repo',
+      default=None,
+      required=True,
+      help='Repo path ex. gcr/oi/projecID/ for output of component build.')
+  args, _ = parser.parse_known_args()
+  return args
+
+
+@dsl.python_component(
```

Comment:
The `@dsl.python_component` is strictly optional (but can be used to override the component name with a human-readable name).

Reply:
Makes sense, removing `@dsl.python_component` from the sample.